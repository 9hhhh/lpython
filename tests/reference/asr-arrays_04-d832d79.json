{
    "basename": "asr-arrays_04-d832d79",
    "cmd": "lfortran --show-asr --no-color {infile} -o {outfile}",
    "infile": "tests/../integration_tests/arrays_04.f90",
    "infile_hash": "d2d40bc4ffd8f6e20e442d0d5b59cdb7e887001405a8f120addf04ab",
    "outfile": null,
    "outfile_hash": null,
    "stdout": "asr-arrays_04-d832d79.stdout",
    "stdout_hash": "737f9f4d50fa5f25935040bd7ff22aa26c5d8c106ba4a77710d1220d",
    "stderr": null,
    "stderr_hash": null,
    "returncode": 0
}