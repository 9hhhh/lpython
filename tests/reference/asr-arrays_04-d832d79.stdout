(TranslationUnit (SymbolTable feecd1f2 {arrays_04: (Program (SymbolTable b1897f4c {a: (Variable b1897f4c a 0 (Real 4 [((Num 1 (Integer 8 [])) (Num 3 (Integer 8 [])))])), abs: (Function (SymbolTable 9996467d {a: (Variable 9996467d a 1 (Real 4 [])), r: (Variable 9996467d r 4 (Integer 4 []))}) abs [(Var 9996467d a)] [(Print () [(Str "abs" (Character 8 []))]) (If (Compare (Var 9996467d a) cmpopType4 (ImplicitCast (Num 0 (Integer 4 [])) cast_kindType1 (Real 4 [])) (Logical 4 [])) [(= (Var 9996467d r) (ImplicitCast (Var 9996467d a) cast_kindType0 (Integer 4 [])))] [(= (Var 9996467d r) (ImplicitCast (UnaryOp unaryopType3 (Var 9996467d a) (Real 4 [])) cast_kindType0 (Integer 4 [])))])] () (Var 9996467d r) ()), b: (Variable b1897f4c b 0 (Real 4 [])), sum: (Function (SymbolTable 70e2a4cd {a: (Variable 70e2a4cd a 1 (Real 4 [(() ())])), i: (Variable 70e2a4cd i 0 (Integer 4 [])), r: (Variable 70e2a4cd r 4 (Integer 4 []))}) sum [(Var 70e2a4cd a)] [(Print () [(Str "sum" (Character 8 []))]) (= (Var 70e2a4cd r) (Num 0 (Integer 4 []))) (DoLoop ((Var 70e2a4cd i) (Num 1 (Integer 4 [])) (FuncCall (Function (SymbolTable 1a7a930f {size: (Variable 1a7a930f size 4 (Integer 4 []))}) size [] [] () (Var 1a7a930f size) ()) [(Var 70e2a4cd a)] [] (Integer 4 [])) ()) [(= (Var 70e2a4cd r) (ImplicitCast (BinOp (ImplicitCast (Var 70e2a4cd r) cast_kindType1 (Real 4 [(() ())])) operatorType0 (ArrayRef (Variable 70e2a4cd a 1 (Real 4 [(() ())])) [(() (Var 70e2a4cd i) ())] (Real 4 [(() ())])) (Real 4 [(() ())])) cast_kindType0 (Integer 4 [])))])] () (Var 70e2a4cd r) ())}) arrays_04 [(= (ArrayRef (Variable b1897f4c a 0 (Real 4 [((Num 1 (Integer 8 [])) (Num 3 (Integer 8 [])))])) [(() (Num 1 (Integer 4 [])) ())] (Real 4 [((Num 1 (Integer 8 [])) (Num 3 (Integer 8 [])))])) (Num 3 (Integer 4 []))) (= (ArrayRef (Variable b1897f4c a 0 (Real 4 [((Num 1 (Integer 8 [])) (Num 3 (Integer 8 [])))])) [(() (Num 2 (Integer 4 [])) ())] (Real 4 [((Num 1 (Integer 8 [])) (Num 3 (Integer 8 [])))])) (Num 2 (Integer 4 []))) (= (ArrayRef (Variable b1897f4c a 0 (Real 4 [((Num 1 (Integer 8 [])) (Num 3 (Integer 8 [])))])) [(() (Num 3 (Integer 4 [])) ())] (Real 4 [((Num 1 (Integer 8 [])) (Num 3 (Integer 8 [])))])) (Num 1 (Integer 4 []))) (= (Var b1897f4c b) (ImplicitCast (FuncCall (Function (SymbolTable 70e2a4cd {a: (Variable 70e2a4cd a 1 (Real 4 [(() ())])), i: (Variable 70e2a4cd i 0 (Integer 4 [])), r: (Variable 70e2a4cd r 4 (Integer 4 []))}) sum [(Var 70e2a4cd a)] [(Print () [(Str "sum" (Character 8 []))]) (= (Var 70e2a4cd r) (Num 0 (Integer 4 []))) (DoLoop ((Var 70e2a4cd i) (Num 1 (Integer 4 [])) (FuncCall (Function (SymbolTable 1a7a930f {size: (Variable 1a7a930f size 4 (Integer 4 []))}) size [] [] () (Var 1a7a930f size) ()) [(Var 70e2a4cd a)] [] (Integer 4 [])) ()) [(= (Var 70e2a4cd r) (ImplicitCast (BinOp (ImplicitCast (Var 70e2a4cd r) cast_kindType1 (Real 4 [(() ())])) operatorType0 (ArrayRef (Variable 70e2a4cd a 1 (Real 4 [(() ())])) [(() (Var 70e2a4cd i) ())] (Real 4 [(() ())])) (Real 4 [(() ())])) cast_kindType0 (Integer 4 [])))])] () (Var 70e2a4cd r) ()) [(Var b1897f4c a)] [] (Integer 4 [])) cast_kindType1 (Real 4 []))) (If (Compare (ImplicitCast (FuncCall (Function (SymbolTable 9996467d {a: (Variable 9996467d a 1 (Real 4 [])), r: (Variable 9996467d r 4 (Integer 4 []))}) abs [(Var 9996467d a)] [(Print () [(Str "abs" (Character 8 []))]) (If (Compare (Var 9996467d a) cmpopType4 (ImplicitCast (Num 0 (Integer 4 [])) cast_kindType1 (Real 4 [])) (Logical 4 [])) [(= (Var 9996467d r) (ImplicitCast (Var 9996467d a) cast_kindType0 (Integer 4 [])))] [(= (Var 9996467d r) (ImplicitCast (UnaryOp unaryopType3 (Var 9996467d a) (Real 4 [])) cast_kindType0 (Integer 4 [])))])] () (Var 9996467d r) ()) [(BinOp (Var b1897f4c b) operatorType1 (ImplicitCast (Num 6 (Integer 4 [])) cast_kindType1 (Real 4 [])) (Real 4 []))] [] (Integer 4 [])) cast_kindType1 (Real 4 [])) cmpopType4 (ConstantReal "1e-5" (Real 4 [])) (Logical 4 [])) [(ErrorStop ())] [])]), size: (Function (SymbolTable 1a7a930f {size: (Variable 1a7a930f size 4 (Integer 4 []))}) size [] [] () (Var 1a7a930f size) ())}) [])
